client/ayon_core/hosts/houdini/api/lib.py

```diff
@@ -998,6 +998,77 @@ def add_self_publish_button(node):
     node.setParmTemplateGroup(template)
 
 
+def get_scene_viewer():
+    """
+    Return an instance of a visible viewport.
+
+    There may be many, some could be closed, any visible are current
+    """
+    panes = hou.ui.paneTabs()
+    panes = [x for x in panes if x.type() == hou.paneTabType.SceneViewer]
+    panes = sorted(panes, key=lambda x: x.isCurrentTab())
+    if panes:
+        return panes[-1]
+
+    return None
+
+
+def sceneview_snapshot(
+        sceneview,
+        filepath="$HIP/thumbnails/$HIPNAME.$F4.jpg",
+        frame_start=None,
+        frame_end=None):
+    """Take a snapshot of your scene view.
+
+    It takes snapshot of your scene view for the given frame range.
+    So, it's capable of generating snapshots image sequence.
+    It works in different Houdini context e.g. Objects, Solaris
+
+    Example:
+    	This is how the function can be used::
+
+        	from ayon_core.hosts.houdini.api import lib
+	        sceneview = hou.ui.paneTabOfType(hou.paneTabType.SceneViewer)
+        	lib.sceneview_snapshot(sceneview)
+
+    Notes:
+        .png output will render poorly, so use .jpg.
+
+        How it works:
+            Get the current sceneviewer (may be more than one or hidden)
+            and screengrab the perspective viewport to a file in the
+            publish location to be picked up with the publish.
+
+        Credits:
+            https://www.sidefx.com/forum/topic/42808/?page=1#post-354796
+
+    Args:
+        sceneview (hou.SceneViewer): The scene view pane from which you want
+                                     to take a snapshot.
+        filepath (str): thumbnail filepath.
```

Comment:
Might be good to elaborate this needs e.g. `$F4` in the path so that it can write out a file per frame.